site-manage/datemanager/tool/src/main/java/org/sakaiproject/datemanager/ThymeleafConfig.java

```diff
@@ -76,4 +81,23 @@ private ITemplateResolver templateResolver() {
 		templateResolver.setTemplateMode(TemplateMode.HTML);
 		return templateResolver;
 	}
+	
+    @Bean
+	public LocaleResolver localeResolver() {
+	    SessionLocaleResolver slr = new SessionLocaleResolver();
+	    slr.setDefaultLocale(Locale.US);
+	    return slr;
+	}
+    
+    @Bean
+	public LocaleChangeInterceptor localeChangeInterceptor() {
+	    LocaleChangeInterceptor lci = new LocaleChangeInterceptor();
+	    lci.setParamName("lang");
+	    return lci;
+	}
+    
+    @Override
+	public void addInterceptors(InterceptorRegistry registry) {
+	    registry.addInterceptor(localeChangeInterceptor());
+	}
```

Comment:
I'm pretty sure you can just replace all this with ResourceLoaderMessageSource.

@Bean
public MessageSource messageSource() {
        ResourceLoaderMessageSource messages = new ResourceLoaderMessageSource();
        messages.setBasename("Messages");
        return messages;
    }

Take a look at:
site/sakai-message-bundle-manager-tool/src/main/java/org/sakaiproject/mbm/ThymeleafConfig.java